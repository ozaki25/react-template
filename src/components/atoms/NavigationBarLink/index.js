import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

function NavigationBarLink({ children, onClick }) {
  return <Link onClick={onClick}>{children}</Link>;
}

export const Link = Button.extend`
  font-size: 30px;
  color: #e70000;
`;

NavigationBarLink.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.node,
};

NavigationBarLink.defaultProps = {
  onClick: null,
  children: null,
};

export default NavigationBarLink;
