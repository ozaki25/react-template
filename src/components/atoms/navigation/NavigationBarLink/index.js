import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const NavigationBarLink = ({ children, onClick }) => <Button onClick={onClick}>{children}</Button>;

const Button = styled.button`
  font-size: 30px;
  color: #e70000;
`;

NavigationBarLink.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
};

NavigationBarLink.defaultProps = {
  onClick: null,
};

export default NavigationBarLink;
