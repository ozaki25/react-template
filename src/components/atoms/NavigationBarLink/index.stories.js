import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';

import NavigationBarLink from '.';

const stories = storiesOf('atoms', module);

stories.add(
  'NavigationBarLink',
  withInfo('ナビゲーションバーの戻るボタンと進むボタンに使うリンク')(() => (
    <NavigationBarLink onClick={action('NavigationBarLink')}>&lt;</NavigationBarLink>
  )),
);
