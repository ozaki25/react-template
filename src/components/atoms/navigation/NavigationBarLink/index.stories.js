import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import NavigationBarLink from '.';

const stories = storiesOf('atoms/navigation', module);

stories.add(
  'NavigationBarLink',
  withInfo('ナビゲーションバーの戻るボタンと進むボタンに使うリンク')(() => (
    <NavigationBarLink onClick={action('NavigationBarLink')}>{text('text', '<')}</NavigationBarLink>
  )),
);
