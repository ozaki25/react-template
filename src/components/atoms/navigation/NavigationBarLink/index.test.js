import React from 'react';
import { snapshot } from 'test/helpers';
import NavigationBarLink from '.';

const props = {
  onClick: jest.fn(),
};

snapshot('NavigationBarLink', <NavigationBarLink {...props}>&lt;</NavigationBarLink>);
