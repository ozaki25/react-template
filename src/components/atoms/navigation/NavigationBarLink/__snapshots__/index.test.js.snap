// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: NavigationBarLink 1`] = `
.c0 {
  font-size: 30px;
  color: #e70000;
}

<NavigationBarLink
  onClick={[Function]}
>
  <styled.button
    onClick={[Function]}
  >
    <button
      className="c0"
      onClick={[Function]}
    >
      &lt;
    </button>
  </styled.button>
</NavigationBarLink>
`;
